### DISCUSSION TITLE
Input of square of adjacency matrix in GCN

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2442] [Q&A]
How can I feed the square of the adjacency matrix of a graph in the GCN  module?

**Comment by rusty1s** [comment]
You can directly pre-compute that, e.g., by utilizing `torch-sparse`:
```
edge_index, _ = torch_sparse.spspmm(edge_index, None, edge_index, None, N, N, N)
```
where `N` denotes the number of nodes.

**Comment by shrimonmuke0202** [comment]
Can I also pass the cube of the adjacency matrix using same way ?

**Reply by rusty1s** [reply]
Yes, but keep in mind that your adjacency matrix can get pretty dense :)

**Comment by shrimonmuke0202** [comment]
Can you give me an example ?
